src/main/java/seedu/address/model/characteristics/Characteristics.java

```diff
@@ -45,12 +45,21 @@ public static boolean isValidCharacteristics(String test) {
     }
 
     /**
-     * Returns true if a given characteristic is contained in the
+     * Returns true if a given string is contained in the
      * characteristics array.
      */
     public boolean containsCharacteristic(String characteristic) {
         return Arrays.stream(characteristicsArr)
-                .anyMatch(c -> c.contains(characteristic));
+                .anyMatch(c -> c.equalsIgnoreCase(characteristic));
+    }
+
+    /**
+     * Returns true if a given {@code Characteristic} contains a characteristic that
+     * is also found in this {@code Characteristic}
+     */
+    public boolean containsCharacteristics(Characteristics other) {
```

Comment:
```suggestion
    public boolean containsAnyGivenCharacteristics(Characteristics other) {
```
Perhaps renaming the function could differentiate it more from `containsCharacteristic`.